code/testapp/src/main/java/com/adobe/marketing/mobile/messagingsample/ScrollingFeedActivity.kt

```diff
@@ -12,58 +12,237 @@
-import com.adobe.marketing.mobile.messaging.Proposition
-import com.adobe.marketing.mobile.messaging.SchemaType
+import com.adobe.marketing.mobile.aepcomposeui.AepUI
+import com.adobe.marketing.mobile.aepcomposeui.SmallImageUI
+import com.adobe.marketing.mobile.aepcomposeui.components.SmallImageCard
+import com.adobe.marketing.mobile.aepcomposeui.style.AepCardStyle
+import com.adobe.marketing.mobile.aepcomposeui.style.AepRowStyle
+import com.adobe.marketing.mobile.aepcomposeui.style.AepTextStyle
+import com.adobe.marketing.mobile.aepcomposeui.style.SmallImageUIStyle
+import com.adobe.marketing.mobile.messaging.ContentCardCallback
+import com.adobe.marketing.mobile.messaging.ContentCardEventObserver
+import com.adobe.marketing.mobile.messaging.ContentCardMapper
+import com.adobe.marketing.mobile.messaging.ContentCardUIProvider
 import com.adobe.marketing.mobile.messaging.Surface
 import com.adobe.marketing.mobile.messagingsample.databinding.ActivityScrollingBinding
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
 
 class ScrollingFeedActivity : AppCompatActivity() {
     private lateinit var binding: ActivityScrollingBinding
+    private lateinit var contentCardUIProvider: ContentCardUIProvider
+    private lateinit var contentCardViewModel: AepContentCardViewModel
+    private lateinit var contentCardCallback: ContentCardCallback
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         binding = ActivityScrollingBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        // retrieve any cached feed propositions
-        var propositions = mutableListOf<Proposition>()
-        val surfaces = mutableListOf<Surface>()
-
         // staging environment - CJM Stage, AJO Web (VA7)
         // surface for content card -
         // mobileapp://com.adobe.marketing.mobile.messagingsample/card/ms
+        val surfaces = mutableListOf<Surface>()
         val surface = Surface("card/ms")
-
-//        val surface = Surface("feeds/apifeed")
         surfaces.add(surface)
-        Messaging.updatePropositionsForSurfaces(surfaces)
-        Messaging.getPropositionsForSurfaces(surfaces) {
-            println("getPropositionsForSurfaces callback contained ${it.entries.size} entry/entries for surface ${surface.uri}")
-            for (entry in it.entries) {
-                for (proposition in entry.value) {
-                    if (isContentCard(proposition)) {
-                        propositions.add(proposition)
+
+        // Initialize the ContentCardUIProvider
+        contentCardUIProvider = ContentCardUIProvider(surface)
+
+        // Initialize the ViewModel
+        contentCardViewModel =
+            ViewModelProvider(this, AepContentCardViewModelFactory(contentCardUIProvider)).get(
+                AepContentCardViewModel::class.java
+            )
```

Comment:
This approach should be OK for our testapp, but, I think we should also provide an example implementation where an Activity as a whole uses a view model  (not entirely creating a new one for the sake of the list) with content cards being fetched inside that viewmodel and retrieved as a KotlinFlow / callbackFlow by the UI. We can discuss more tomorrow.